assets/src/blocks/CarouselHeader/useSlides.js

```diff
@@ -0,0 +1,65 @@
+import { useState } from '@wordpress/element';
+
+export const useSlides = (slides, slidesRef) => {
```

Comment:
The logic of this hook seems to be copied from https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/b2ad5d4448aa089a887faa2b7d85af95e781812e/assets/src/blocks/Gallery/GalleryCarousel.js#L10-L14

Probably better to make that component use the hook too while creating it, then we won't have duplicate logic and can actually improve things in one place.

Reply:
If both components can't use the same hook, that probably means the hook has too many responsibilities.